velox/exec/tests/AggregationTest.cpp

```diff
@@ -503,6 +516,23 @@ TEST_F(AggregationTest, missingFunctionOrSignature) {
       "Supported signatures: (smallint,varchar) -> tinyint -> bigint.");
 }
 
+
+TEST_F(AggregationTest, castVarcharToTimestamp) {
+
+  testVarchartoTimestamp(Timestamp(-946684800, 0), "1940-01-02 00:00:00");
```

Comment:
I have used similar test cases that were added for timestamp to string UTs.
https://github.com/facebookincubator/velox/blob/650eb6a24a9fa68cdae77d72d86d79fa05cfd29e/velox/type/tests/TimestampTest.cpp#L475C3-L508C2